tests/providers/salesforce/hooks/test_salesforce.py

```diff
@@ -48,8 +49,41 @@ def test_get_conn_exists(self):
 
         assert self.salesforce_hook.conn.return_value is not None
 
+    @parameterized.expand(
+        [
+            (
+                "all extras",
+                '''
+                {
+                    "extra__salesforce__client_id": "my_client",
+                    "extra__salesforce__consumer_key": "",
+                    "extra__salesforce__domain": "test",
+                    "extra__salesforce__instance": "",
+                    "extra__salesforce__instance_url": "",
+                    "extra__salesforce__organization_id": "",
+                    "extra__salesforce__private_key": "",
+                    "extra__salesforce__private_key_file_path": "",
+                    "extra__salesforce__proxies": "",
+                    "extra__salesforce__security_token": "token",
+                    "extra__salesforce__version": "42.0"
+                }
+                ''',
+            ),
+            (
+                "required extras",
+                '''
+                {
+                    "extra__salesforce__client_id": "my_client",
+                    "extra__salesforce__domain": "test",
+                    "extra__salesforce__security_token": "token",
+                    "extra__salesforce__version": "42.0"
+                }
+                ''',
+            ),
+        ]
```

Comment:
I see that you have added parameterization in 4 places, removing the "empty string" extras.

I think we can simplify this and be more direct about what we're testing.

What do you think about the following.

Add one test that verifies that when extra params are supplied as empty string, the call converts those params to `None`

Then, for all of the tests you have parameterized, instead of parameterizing, just chop down the params to only the non-empty-string params.  We'll know they will be converted to `None` since you will add the test suggested above, so they no longer need to be included in those 4 tests.  This leaves just your "required params" sets.

Reply:
I agree. That sounds good actually. A lot less lines of code.